src/nvim/api/window.c

```diff
@@ -63,6 +63,10 @@ void nvim_win_set_buf(Window window, Buffer buffer, Error *err)
     return;
   }
 
+  if (win->w_buffer == buf) {
+    return;
```

Comment:
probably makes sense to combine with the condition on line 62?

same for the other case.